Setup: Improve Ruby/Jekyll instructions

http://carpentries.github.io/lesson-example/setup.html#optional-jekyll-setup-for-lesson-development

I have tried to follow the setup instructions and have run into this problem which isn't apparent from the instructions: https://stackoverflow.com/questions/17618113/the-command-rbenv-install-is-missing

Instead, as I previously have done, I have just installed Jekyll using 
```
sudo apt-get install build-essential
sudo apt install ruby2.5-dev
sudo gem2.5 install jekyll
```
and this seems to work fine (Linux Mint 19). Presumably the above commands would also be required if I'm using rbenv?

I'm writing a [PR skill-up workshop](https://gcapes.github.io/swc-pr-tutorial/), and part of it includes viewing changes locally. I think the jekyll setup instructions will be a stumbling block for many people. I have certainly had a lot of difficulty with this before, and the [Jekyll site](https://jekyllrb.com) makes it seem so easy 
> get up and running in seconds

I'd prefer to point to the documentation in this repo rather than produce my own how-to, but currently it seems like the instructions aren't complete enough (for Linux at least - I've not tested this on Windows yet).

Thanks Maxim.

Installing `ruby-build` does look to be all that is required. I uninstalled `ruby2.5-dev` first.

However, the `rbenv` command didn't work.

```
$ rbenv install 2.5.3
ruby-build: definition not found: 2.5.3

The following versions contain `2.5.3' in the name:
  rbx-2.5.3

See all available versions with `rbenv install --list'.
```
So then i did
```
rbenv install rbx-2.5.3
```

```
Downloading rubinius-2.5.3.tar.bz2...
-> https://dqw8nmjcqpjn7.cloudfront.net/9af4d6e9d1e78a586579c86b9eb9a082cb863885d4a7cf33989d73280461e5fc
Installing rubinius-2.5.3...

WARNING: rubinius-2.5.3 is past its end of life and is now unsupported.
It no longer receives bug fixes or critical security updates.


BUILD FAILED (LinuxMint 19.1 using ruby-build 20170726)

Inspect or clean up the working tree at /tmp/ruby-build.20190510105150.25481
Results logged to /tmp/ruby-build.20190510105150.25481.log

Last 10 log lines:
/tmp/ruby-build.20190510105150.25481/rubinius-2.5.3/vendor/bundle/ruby/2.5.0/extensions/x86_64-linux/2.5.0/rubinius-melbourne-2.3.1.0/gem_make.out

An error occurred while installing rubinius-melbourne (2.3.1.0), and Bundler
cannot continue.
Make sure that `gem install rubinius-melbourne -v '2.3.1.0' --source
'https://rubygems.org/'` succeeds before bundling.

In Gemfile:
  rubinius-build_tools was resolved to 2.0.0, which depends on
    rubinius-melbourne

```

So I abandoned that and installed jekyll as before using ruby 2.5.1.

```
 $ sudo gem install jekyll
Fetching: jekyll-3.8.5.gem (100%)
Successfully installed jekyll-3.8.5
Parsing documentation for jekyll-3.8.5
Installing ri documentation for jekyll-3.8.5
Done installing documentation for jekyll after 1 seconds
1 gem installed
```

I would suggest checking these installation instructions for the OS covered. If it's important to use the same version of Ruby as GitHub, the instructions should cover how to do this. However I've not yet encountered a problem using a different version of Ruby.

If the instructions are incomplete, or incorrect, contributors will have to be very determined to resolve problems on their own, and simply won't preview their changes.

--- a/setup.md
+++ b/setup.md
@@ -167,10 +167,12 @@ new lesson is `data-cleanup`.
 2.  Other lessons' `setup.md` include full installation instructions organized by OS
     (following the model of the workshop template home page).
 
-## (Optional) Jekyll Setup for Lesson Development
+## Jekyll Setup for Lesson Development
 
-If you want to preview changes on your own machine before pushing them to GitHub,
-you must install the software described below.
+Though not essential, it is desirable to be able to preview changes on your own machine
+before pushing them to GitHub.
+
+In order to preview changes locally, you must install the software described below.
 
 1.  **[Ruby](https://www.ruby-lang.org/en/downloads/)**.
 

--- a/setup.md
+++ b/setup.md
@@ -206,7 +206,7 @@ you must install the software described below.
     For more information, see [the Ruby installation guidelines][ruby-install-guide].
 
 2.  **[RubyGems][rubygems]**
-    is a tool which manages Ruby packages. It should be installed along with Ruby and you can
+    is a tool which manages Ruby packages. It should have been installed along with Ruby and you can
     test your installation by running
 
     ~~~

--- a/setup.md
+++ b/setup.md
@@ -203,6 +203,9 @@ you must install the software described below.
     ~~~
     {: .language-bash}
 
+    Windows users should select the start menu option `Start command prompt with Ruby`
+    to execute the above command.
+
     For more information, see [the Ruby installation guidelines][ruby-install-guide].
 
 2.  **[RubyGems][rubygems]**

--- a/setup.md
+++ b/setup.md
@@ -176,8 +176,11 @@ In order to preview changes locally, you must install the software described bel
 
 1.  **[Ruby](https://www.ruby-lang.org/en/downloads/)**.
 
-    **Linux/macOS**: Ruby is usually included with Linux and macOS. However, to reliably render
-    lessons the way GitHub does, we have to use the same version of Ruby as GitHub. Currently,
+    **Linux/macOS**: Ruby is usually included with Linux and macOS ---
+    this is usually sufficient to preview your changes locally.
+
+    If you aren't able to preview reliably with your default Ruby installation,
+    you can use the same version of Ruby as GitHub. Currently,
     GitHub uses Ruby 2.5.3. In order to install Ruby 2.5.3 on Linux and macOS, we recommend using
     [rbenv](https://github.com/rbenv/rbenv):
 

--- a/setup.md
+++ b/setup.md
@@ -171,7 +171,6 @@ new lesson is `data-cleanup`.
 
 If you want to preview changes on your own machine before pushing them to GitHub,
 you must install the software described below.
-Julian Thilo wrote instructions for [installing Jekyll on Windows][jekyll-windows].
 
 1.  **[Ruby](https://www.ruby-lang.org/en/downloads/)**.
 

